db/db_impl/db_impl.cc

```diff
@@ -5631,6 +5630,91 @@ Status DBImpl::CreateColumnFamilyWithImport(
   return status;
 }
 
+Status DBImpl::ClipDB(ColumnFamilyHandle* column_family, const Slice& begin_key,
+                      const Slice& end_key) {
+  assert(column_family);
+  Status status;
+
+  {
+    // Lock db mutex
+    InstrumentedMutexLock l(&mutex_);
```

Comment:
We should only need mutex_ around `cfd->current()->GetSstFilesBoundaryKeys(&smallest_user_key, &largest_user_key);` now.